src/libraries/System.Text.RegularExpressions/tests/UnitTests/RegexReductionTests.cs

```diff
@@ -495,6 +498,8 @@ public void PatternsReduceIdentically(string actual, string expected)
         [InlineData("a*(?(xyz)acd|efg)", "(?>a*)(?(xyz)acd|efg)")]
         [InlineData("a*(?(xyz)bcd|afg)", "(?>a*)(?(xyz)bcd|afg)")]
         [InlineData("a*(?(xyz)bcd)", "(?>a*)(?(xyz)bcd)")]
+        // Different prefixes on alternation branches
+        [InlineData("^abcd|$abce", "^abcd|^abce")]
```

Comment:
is it significant here that it's $ (is this multiline?)

Reply:
Just that it's a different anchor from ^